### PR TITLE
Add an IsDead check to prevent crashes in survival01

### LINKED ISSUE #11662: Uncaught CLR exception at Lua->CLR boundary in Red Alert mission Survival 1
Fatal Lua Error: [string "BindingSupport.lua"]:30: Uncaught CLR exception at Lua->CLR boundary: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Eluant.LuaException: [string "BindingSupport.lua"]:30: Uncaught CLR exception at Lua->CLR boundary: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Attempted to get trait from destroyed object (e1 699 (not in world))
  at OpenRA.TraitDictionary.CheckDestroyed (OpenRA.Actor actor) <0x40c8fc30 + 0x00093> in <filename unknown>:0 
  at OpenRA.TraitDictionary.GetOrDefault[T] (OpenRA.Actor actor) <0x40c919f0 + 0x0001f> in <filename unknown>:0 
  at OpenRA.Actor.TraitOrDefault[T] () <0x40c919a0 + 0x0003b> in <filename unknown>:0 
  at OpenRA.Mods.Common.Scripting.TriggerGlobal.GetScriptTriggers (OpenRA.Actor a) <0x40d6d230 + 0x00027> in <filename unknown>:0 
  at OpenRA.Mods.Common.Scripting.TriggerGlobal.OnIdle (OpenRA.Actor a, Eluant.LuaFunction func) <0x40d70b00 + 0x0001b> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000a1> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000ef> in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) <0x7fa73e3e3cf0 + 0x0002a> in <filename unknown>:0 
  at OpenRA.Scripting.ScriptMemberWrapper.Invoke (Eluant.LuaVararg args) <0x40d560a0 + 0x004bc> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000a1> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000ef> in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) <0x7fa73e3e3cf0 + 0x0002a> in <filename unknown>:0 
  at Eluant.LuaRuntime+MethodWrapper.Invoke (System.Object[] parms) <0x40d56060 + 0x00035> in <filename unknown>:0 
  at Eluant.LuaRuntime.MakeManagedCall (IntPtr state, Eluant.MethodWrapper wrapper) <0x40d54750 + 0x0108b> in <filename unknown>:0 
  at Eluant.LuaRuntime.Call (IList`1 args) <0x40d48890 + 0x002bf> in <filename unknown>:0 
  at Eluant.LuaRuntime.Call (Eluant.LuaFunction fn, IList`1 args) <0x40d4dbf0 + 0x0004f> in <filename unknown>:0 
  at Eluant.LuaFunction.Call (IList`1 args) <0x40d4dbb0 + 0x0002b> in <filename unknown>:0 
  at Eluant.LuaFunction.Call (Eluant.LuaValue[] args) <0x40d4db80 + 0x00013> in <filename unknown>:0 
  at OpenRA.Mods.Common.Scripting.UtilsGlobal.Do (Eluant.LuaValue[] collection, Eluant.LuaFunction func) <0x40d597c0 + 0x0007f> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000a1> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000ef> in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) <0x7fa73e3e3cf0 + 0x0002a> in <filename unknown>:0 
  at OpenRA.Scripting.ScriptMemberWrapper.Invoke (Eluant.LuaVararg args) <0x40d560a0 + 0x004bc> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000a1> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x7fa73e5ab9e0 + 0x000ef> in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) <0x7fa73e3e3cf0 + 0x0002a> in <filename unknown>:0 
  at Eluant.LuaRuntime+MethodWrapper.Invoke (System.Object[] parms) <0x40d56060 + 0x00035> in <filename unknown>:0 
  at Eluant.LuaRuntime.MakeManagedCall (IntPtr state, Eluant.MethodWrapper wrapper) <0x40d54750 + 0x0108b> in <filename unknown>:0 
  at OpenRA.Scripting.ScriptContext.FatalError (System.String message) [0x00000] in <filename unknown>:0 
  at OpenRA.Mods.Common.Scripting.TriggerGlobal+<AfterDelay>c__AnonStorey0.<>m__0 () [0x00000] in <filename unknown>:0 
  at OpenRA.Effects.DelayedAction.<Tick>m__0 (OpenRA.World w) [0x00000] in <filename unknown>:0 
  at OpenRA.World.Tick () [0x00000] in <filename unknown>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x00000] in <filename unknown>:0 
  at OpenRA.Game.LogicTick () [0x00000] in <filename unknown>:0 
  at OpenRA.Game.Loop () [0x00000] in <filename unknown>:0 
  at OpenRA.Game.Run () [0x00000] in <filename unknown>:0 
  at OpenRA.Program.Run (System.String[] args) [0x00000] in <filename unknown>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0

### PR DESCRIPTION
**Pull request by abcdefg30**
Fixes #11662. (Probably)

### CODE REVIEW

File: mods/ra/maps/survival01/survival01.lua [hunk 1]
```diff
@@ -66,45 +66,38 @@ SovietBuildings = { Barrack1, SubPen, RadarDome, AdvancedPowerPlant1, AdvancedPo
...
-								end)
-							end
-						end)
-					end
-				end)
-			end
-		end)
+		if not unit.IsDead then
+			Trigger.OnIdle(unit, function()
+				local bool = Utils.All(units, function(unit) return unit.IsIdle end)
```
Comment (pchote) [inline mods/ra/maps/survival01/survival01.lua]: Why does any one unit being idle set up the hunt trigger for *all* units?  This smells quite bogus.

Comment (abcdefg30) [inline mods/ra/maps/survival01/survival01.lua]: It only sets it up when all other units are `Idle` as well. I already thought about introducing an `Trigger.OnAllIdle` as followup.

However, even if this is bogus: This is not part of the PR, I just moved already existing code that works fine.

**Comment by obrakmann** [comment]
Looks fine, :+1:

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/pull/11666/files?w=1

**Pull request merged**